ReactiveSwift-UIExamples.playground/Pages/TransactionalProperty.xcplaygroundpage/Contents.swift

```diff
@@ -0,0 +1,133 @@
+import ReactiveSwift
+import Result
+import UIKit
+import PlaygroundSupport
+
+final class User {
+	let username = MutableProperty("example")
+}
+
+final class ViewModel {
+	struct FormError: Error {
+		let reason: String
+
+		static let invalidEmail = FormError(reason: "The address must end with `@reactivecocoa.io`.")
+		static let mismatchEmail = FormError(reason: "The e-mail addresses do not match.")
+	}
+
+	let email: TransactionalProperty<String, FormError>
+	let emailConfirmation: TransactionalProperty<String, FormError>
+	let termsAccepted: MutableProperty<Bool>
+	let reasons: Signal<String, NoError>
+
+	let submit: Action<(), String, NoError>
+
+	init(_ user: User) {
+		termsAccepted = MutableProperty(false)
+
+		// Validation for `email`.
+		email = user.username
+			.map(forward: { !$0.isEmpty ? "\($0)@reactivecocoa.io" : "" },
+			     attemptBackward: { Result($0.stripSuffix("@reactivecocoa.io"),
+			                               failWith: .invalidEmail) })
+
+		// Validation for `emailConfirmation`.
+		emailConfirmation = MutableProperty("")
+			.validate(with: email) { input, currentEmail in
+				return input == currentEmail ? .success(()) : .failure(.mismatchEmail)
+			}
+
+		// The aggregate of latest validation results of all text fields.
+		let validationResults = Property.combineLatest(email.validations,
+		                                    emailConfirmation.validations)
+
+		// The validation state of the entire form.
+		let isValid = validationResults
+			.combineLatest(with: termsAccepted)
+			.map { $0.0.value != nil && $0.1.value != nil && $1 }
+
+		let state = isValid.combineLatest(with: user.username)
+
+		// Aggregate latest failures into stream of strings.
+		reasons = validationResults.signal
+			.skip(first: 1)
+			.map {
+				return [$0, $1]
+					.flatMap { $0.error?.reason }
+					.joined(separator: "\n")
+			}
+
+		// The action to be invoked when the submit button is pressed.
+		// It enables only if all the controls have passed their validations.
+		submit = Action(state: state, enabledIf: { $0.0 }) { state, _ in
+			return SignalProducer { observer, disposable in
+				observer.send(value: state.1)
+				observer.sendCompleted()
+			}
+		}
+	}
+}
+
+final class ViewController {
```

Comment:
Also, what is the reasoning behind making the ViewModel a class? (I am quite curious, since I struggled with this decision)